spec/controllers/accounts/features_controller_spec.rb

```diff
@@ -0,0 +1,238 @@
+        expect(assigns(:account).id).to eq account.id
+        expect(assigns(:account).name).to eq account.name
+      end
+      it 'assigns @feature' do
+        expect(assigns(:feature)).to be_a_kind_of Feature
+        expect(assigns(:feature).id).to eq feature.id
+        expect(assigns(:feature).name).to eq feature.name
+      end
+    end
+  end
+
+  describe "PATCH 'update'" do
+    let(:feature_update_request) do
+      uri = Regexp.new "#{Feature.api_root}/features/#{feature.id}"
+      WebMock.stub_request(:patch, uri)
+        .with(headers: { 'Http-Authorization' => Feature.api_token })
+        .to_return(status: 200, body: feature_response)
+    end
+
+    context 'anonymous user' do
+      before do
+        feature_find_request
+        feature_update_request
+        patch :update, id: feature.id, feature: { name: 'hello' }, account_id: account.id
+      end
+      it 'redirects to account_feature_path(@account, @feature)' do
+        expect(response.status).to be 302
+        expect(response).to redirect_to(account_feature_path(assigns(:account), assigns(:feature)))
+      end
+      it 'assigns @account' do
+        expect(assigns(:account)).to be_a_kind_of Account
+        expect(assigns(:account).id).to eq account.id
+        expect(assigns(:account).name).to eq account.name
+      end
+      it 'assigns @feature' do
+        expect(assigns(:feature)).to be_a_kind_of Feature
+        expect(assigns(:feature).id).to eq feature.id
+        expect(assigns(:feature).name).to eq feature.name
+      end
+      it 'makes the feature update request' do
+        expect(feature_update_request).to have_been_requested
+      end
+      it 'makes the feature find request' do
+        expect(feature_find_request).to have_been_requested
+      end
+      it 'makes the account find request' do
+        expect(account_find_request).to have_been_requested
+      end
+    end
+  end
+
+  describe "DELETE 'destroy'" do
+    let(:feature_delete_request) do
+      uri = Regexp.new "#{Feature.api_root}/features/#{feature.id}"
+      WebMock.stub_request(:delete, uri)
+        .with(headers: { 'Http-Authorization' => Feature.api_token })
+        .to_return(status: 200, body: feature_response)
+    end
+
+    context 'anonymous user' do
+      before do
+        feature_find_request
+        feature_delete_request
+        delete :destroy, id: feature.id, account_id: account.id
+      end
+      it 'redirects to account_features_path(@account)' do
+        expect(response.status).to be 302
+        expect(response).to redirect_to(account_features_path(assigns(:account)))
+      end
+      it 'makes the feature delete request' do
+        expect(feature_delete_request).to have_been_requested
+      end
+      it 'makes the account find request' do
+        expect(account_find_request).to have_been_requested
+      end
+      it 'makes the feature find request' do
+        expect(feature_find_request).to have_been_requested
+      end
```

Comment:
Pretty quick but complete-ish pass at controller specs. There's lots of repetition in setup between various controllers and view specs. Should be refactored down later on.